master/buildbot/www/hooks/github.py

```diff
@@ -114,9 +114,17 @@ def _get_payload(self, request):
             mac = hmac.new(unicode2bytes(self._secret),
                            msg=unicode2bytes(content),
                            digestmod=sha1)
-            # NOTE: hmac.compare_digest should be used, but it's only available
-            # starting Python 2.7.7
-            if mac.hexdigest() != hexdigest:
+
+            def _cmp(a, b):
+                try:
+                    # try the more secure compare_digest() first
+                    from hmac import compare_digest
```

Comment:
we don't support python 2.6, so that's fine to always use  compare_digest

Reply:
Hmm. Are Python 2.7.0~2.7.6 unsupported, either?

Reply:
well, we don't go to that precision, indeed. But we don't test python minor versions, so I don't . I guess lets keep it that way, as it is not much of a burden. Stats show that there are around 111 unique installs with python 2.7.5..